homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -394,6 +401,16 @@ def __post_init__(self) -> None:
             )
         pipeline_data.pipeline_runs[self.pipeline.id][self.id] = PipelineRunDebug()
 
+        # Directory to save audio for each pipeline run.
+        # Configured in YAML for assist_pipeline.
+        debug_recording_dir = self.hass.data.get(DATA_CONFIG, {}).get(
```

Comment:
key should always exist.

```suggestion
        debug_recording_dir = self.hass.data[DATA_CONFIG].get(
```